.github/workflows/integration.yml

```diff
@@ -51,3 +51,7 @@ jobs:
       - name: Remove test logs
         if: always()
         run: rm -rf /tmp/goth-tests
+
+      # Only relevant for self-hosted runners
+      - name: Remove poetry virtual env
+        run: poetry env remove python3.8
```

Comment:
Perhaps let's add a comment here with the relevant issue in poetry (https://github.com/python-poetry/poetry/issues/3208), so that this step can remove all envs once that change is merged.